app/helpers/payment_details_helper.rb

```diff
@@ -98,4 +99,8 @@ def country_code_for(country_name)
 
     merchant_countries.find { |country| country[0] == country_name }&.dig(1)
   end
+
+  def ip_address
+    request&.remote_ip
```

Comment:
why `&`? Can `request` be `nil`?

Reply:
Yeah... I'm not sure about it. In a normal processing, where the helper is used in some view that is rendered by a controller, it would never be `nil` I guess.

I noticed this in a test, but I later fixed it by mocking the `ActionDispatch::Request`. But I decided to keep this, just in case I don't know... this helper is used in some other test where it's not mocked, or maybe the helper is used outside of the controller. Not sure if that can ever happen, but if it does, I'd just prefer this not to fail with an exception :grimacing:

Reply:
`request` should always exist as part of the rails in a controller action, no?

Reply:
> `request` should always exist as part of the rails in a controller action, no?

Yes, I think so. What I was saying is - I don't know if it's possible that the helper is used outside of the controller action.

But if you think it's weird to use safe navigation here, I can remove it, and hope it will never fail :)